public/app/plugins/datasource/loki/queryUtils.ts

```diff
@@ -364,15 +372,17 @@ export function combineResponses(currentResult: DataQueryResponse | null, newRes
       currentResult.data.push(newFrame);
       return;
     }
-    combine(currentFrame, newFrame);
+    combine(new MutableDataFrame(currentFrame), newFrame);
   });
 
   return currentResult;
 }
 
-function combine(dest: DataQueryResponseData, source: DataQueryResponseData) {
-  for (let j = 0; j < source.fields[0].values.length; j++) {
+function combine(dest: MutableDataFrame, source: MutableDataFrame) {
```

Comment:
if you can avoid `MutableDataFrame` that is better -- it has a LOT of overhead to make things feel simple/easy, but if not required will avoid a bunch

Reply:
Thanks! Removed in https://github.com/grafana/grafana/pull/62767/commits/b1762240bb290661abef255995a27403871fae1d

Reply:
Thanks for the info @ryantxu!